### PR TITLE
Improve visibility of startup problems

### LINKED ISSUE #1873: Improve visibility of plugin startup problems
If the plugin detects a fatal error condition and exits, it gets restarted by Docker because of the `--restart=always`.  However this situation is not made very clear to the user.

For example, a bad option will cause an immediate exit:

````
$ weave launch-plugin --foo
7c4094d284ecec15addf8a2e800cc42a2509db51f4675e5a7479eeae45a03731
````

No hint here, and worse, `weave status` gives a normal report:

````
$ weave status
[...]
        Service: plugin
     DriverName: weave
````

Compare:

````
$ weave launch-router --foo
The weave container has died. Consult the container logs for further details.
````

Note #1814 fixed some related things in the weave script.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #1873

Also improves behaviour in the case that the user has set a restart policy on weaver or weaveproxy, and then encountered a problem.

### CODE REVIEW

**Comment by bboreham** [comment]
Test 640 failed because the container is detected as 'restarting' and `container_id` barfs.  This means there was always a race here, whether the restarting container is actually in a state to have a network attached.

That race is now covered by #1561. But not on this branch.

**Comment by rade** [comment]
`State.Status` is new in Docker 1.9. We get away with using it in the current `check_not_running` through sheer luck. Not so in this new code.

**Comment by bboreham** [comment]
Added `wait_for_status` to `launch_plugin`, and rebased

File: prog/plugin/main.go [hunk 1]
```diff
@@ -168,3 +170,19 @@ func createNetwork(dockerClient *docker.Client, networkName, address string) err
...
+	if err != nil {
+		Log.Fatalf("ListenAndServeStatus failed: %s", err)
+	}
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "ok")
+	})
+	if err := (&http.Server{Handler: handler}).Serve(listener); err != nil {
+		Log.Fatalf("ListenAndServeStatus failed: %s", err)
+	}
+}
```
Comment (rade) [inline prog/plugin/main.go]: this is c&p from the proxy. And the socket removal+listening bit is also duplicated in the plugin. And we really don't want those `Fatalf`, at least not in the first part.

**Comment by rade** [comment]
This is getting a bit too scary for 1.4, IMO.

**Comment by bboreham** [comment]
Replaced by #1941 on master

**Pull request closed without merging**